runtime/src/asset.rs

```diff
@@ -345,12 +388,26 @@ decl_module! {
                 name,
                 total_supply,
                 owner_did: did,
-                divisible: divisible
+                divisible,
+                asset_type: asset_type.clone(),
             };
-
             <Tokens<T>>::insert(&ticker, token);
             <BalanceOf<T>>::insert((ticker.clone(), did), total_supply);
-            Self::deposit_event(RawEvent::IssuedToken(ticker, total_supply, did, divisible));
+            Self::deposit_event(RawEvent::IssuedToken(
+                ticker.clone(),
+                total_supply,
+                did,
+                divisible,
+                asset_type,
+            ));
+            for (typ, val) in identifiers {
+                <Identifiers>::insert((ticker.clone(), typ.clone()), val.clone());
+                Self::deposit_event(RawEvent::IdentifierRegistered(
+                    ticker.clone(),
+                    typ.clone(),
+                    val.clone()
```

Comment:
Since these values are not used anymore, we should be able to avoid cloning them.

Reply:
They have to be cloned because they are used multiple times within the `for` loop.

Reply:
Fixed that while rebasing.